### ISSUE TITLE
Allow configure ApiV4Auth Timestamp value. 

### ISSUE THREAD

**Issue by timaiv** [gateio/gateapi-csharp#31]
We have made api proxy.
For safety, we need to take the timestamp value that was calculated on api proxy client machine and passed to proxy, since the request itself can take several seconds.

https://github.com/gateio/gateapi-csharp/blob/7d6881744da1839d02874b3e2b997c09bc540b6b/src/Io.Gate.GateApi/Client/ApiClient.cs#L348

**Comment by revilwang** [comment]
I don't think this will for you. If your api client machine has a different timestamp with your proxy machine, you need to recalculate the signature. Plus, several seconds should not worry you. The server allows such time gap.

**Comment by timaiv** [comment]
> I don't think this will for you. If your api client machine has a different timestamp with your proxy machine, you need to recalculate the signature. Plus, several seconds should not worry you. The server allows such time gap.
Hi!!

I think you misunderstood me.
Let's say the request window is 10 seconds, (which I would also like to set as in mexc, which unfortunately is not supported).

Simple example.
1) The user made a request at 10m:10s
2) Due to network problems, gateways, retries, slow server etc, api proxy got request (developer code executing) at 10m:21sec (after real 11 seconds).
3) Now lib in api proxy creates timestamp for window (10m:21-10m:31), but for security this request is too late and should not be executed.

And this is a very simple example. In a production, the timestamp transfer can be sent multiple times (similar to X-Forwarded-For).

Since there is no way to set the timestamp here, I will have to duplicate the mechanism between the client and the proxy and create my own time and signature.
